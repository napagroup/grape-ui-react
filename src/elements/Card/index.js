export * from './component';
export * from './styled';
