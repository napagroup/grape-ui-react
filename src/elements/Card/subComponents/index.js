export * from './CardActions';
export * from './CardBody';
export * from './CardHeader';
export * from './CardInner';
export * from './CardRichMedia';
export * from './CardSecondaryMedia';
