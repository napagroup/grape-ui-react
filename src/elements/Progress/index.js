export * from './getProgress';
export * from './styled';
