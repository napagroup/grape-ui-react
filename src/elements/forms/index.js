export * from './CheckboxField';
export * from './DateField';
export * from './Form';
export * from './SelectField';
export * from './TextField';
