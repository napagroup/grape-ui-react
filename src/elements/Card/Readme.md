The `Card` component is intended to convey content and actions relating to a single idea.  In `grape-ui`, you can use `Card` and pass in contents via its props, its children, or via sub components.

#### Basic Example

```jsx is Markdown
import { Card } from 'src/elements/Card';
import { Box, Flex } from 'src/elements/grid';
import { CodeBlock } from 'src/elements/typography';

const baseCardStyleProps = {
  m: 1,
  maxWidth: 300,
  width: 1,
};

const containerStyleProps = {
  alignItems: 'center',
  background: '#fdf4fb',
  flexDirection: ['column', 'row'],
  justifyContent: 'space-evenly',
  p: [2, null, 3],
};

<Flex {...containerStyleProps}>
  <Box {...baseCardStyleProps}>
    <CodeBlock code="// Passed in through Props" />
    <Card
      cardSubtitle="Secondary text"
      cardTitle="Card title"
      {...baseCardStyleProps}
    />
  </Box>
  <Box {...baseCardStyleProps}>
    <CodeBlock code="// Using subComponents" />
    <Card {...baseCardStyleProps}>
      <Card.Title>
        Card title
      </Card.Title>
      <Card.Subtitle>
        Secondary text
      </Card.Subtitle>
    </Card>
  </Box>
</Flex>
```

#### Kitchen Sink

```jsx is Markdown
import { Button } from 'src/elements/Button';
import { Card } from 'src/elements/Card';
import {
  Box,
  Flex,
} from 'src/elements/grid';
import { Image } from 'src/elements/Image';
import {
  CodeBlock,
  Paragraph,
} from 'src/elements/typography';

const getRandomBill = (min, max, ratio) => {
  const numberX = Math.floor(Math.random() * (max - min) + min);
  const numberY = Math.floor(numberX / ratio);
  return `http://www.fillmurray.com/${numberX}/${numberY}`;
};

const ExampleCardActions01 = (
  <Button bg="brandPrimary" bgHoverColor="brandPrimary.light" color="white" m={0}>Action</Button>
);

const ExampleCardActions02 = (
  <Button>üèåÔ∏è‚Äç‚ôÄÔ∏è</Button>
);

const ExampleCardBody01 = (
  <Paragraph ellipsis={5} mb="0">
    What an incredible Cinderella story. This unknown comes outta nowhere to lead the pack at Augusta. He‚Äôs at the final hole. He‚Äôs about 455 yards away, he‚Äôs gonna hit about a two iron, I think ‚Ä¶ (Carl reels back and swats the head off of a mum. Petals fly like confetti) Boy, he got all of that. The crowd is standing on its feet here at Augusta. The normally reserved Augusta crowd is going wild ‚Ä¶ (he pauses as he notices some golfers coming) for this young Cinderella who‚Äôs come out of nowhere. He‚Äôs got about 350 yards left. He‚Äôs going to hit about a five iron, it looks like, don‚Äôt you think? (Carl pulls the grass whip back to demolish the next mum) He‚Äôs got a beautiful backswing ‚Ä¶ That‚Äôs ‚Ä¶ Oh! He got all of that one! He‚Äôs gotta be pleased with that. The crowd is just on its feet here. He‚Äôs a Cinderella boy, tears in his eyes, I guess, as he lines up this last shot. And he‚Äôs got about 195 yards left, and he‚Äôs got a, it looks like he‚Äôs got about an eight iron. This crowd has gone deadly silent. Cinderella story, out of nowhere, former greenskeeper, now about to become the Masters champion. (Carl reels back one last time and ‚Äî Swat! ‚Äî blasts the third mum to smithereens) It looks like a mirac . . . It‚Äôs in the hole! IT‚ÄôS IN THE HOLE!!!
  </Paragraph>
);

const ExampleCardImage01 = (
  <Image
    alt="Placeholder photo from FillMurray.com"
    display="block" // be sure to mark this as block or else the image can cause unwanted spacing on the bottom of the card
    src={getRandomBill(400, 1000, 16 / 9)}
  />
);

const ExampleCardImage02 = (
  <Image
    alt="Placeholder photo from FillMurray.com"
    display="block" // be sure to mark this as block or else the image can cause unwanted spacing on the bottom of the card
    src={getRandomBill(400, 1000, 4 / 3)}
  />
);

const ExampleCardThumbnail01 = (
  <Image
    alt="Placeholder photo from FillMurray.com"
    borderRadius="50%"
    display="block" // be sure to mark this as block or else the image can cause unwanted spacing on the bottom of the card
    src={getRandomBill(40, 100, 1)}
    width={40}
  />
);

const exampleCardPadding = 3;

const baseBoxStyleProps = {
  m: 1,
  maxWidth: 300,
  width: 1,
}

const baseCardStyleProps = {
  ...baseBoxStyleProps,
  showProgress: true,
};

const containerStyleProps = {
  alignItems: 'center',
  background: '#fdf4fb',
  flexWrap: 'wrap',
  justifyContent: 'space-evenly',
  p: exampleCardPadding,
};

<Flex {...containerStyleProps}>
  <Box {...baseBoxStyleProps}>
    <CodeBlock code="// Passed in through Props" />
    <Card
      cardActions={ExampleCardActions01}
      cardActionsRight={ExampleCardActions02}
      cardBody={ExampleCardBody01}
      cardRichMedia={ExampleCardImage02}
      cardSubtitle="Secondary text"
      cardThumbnail={ExampleCardThumbnail01}
      cardTitle="Card title"
      {...baseCardStyleProps}
    />
  </Box>
  <Box {...baseBoxStyleProps}>
    <CodeBlock code="// Using subComponents" />
    <Card cardPadding={0} pb={exampleCardPadding} {...baseCardStyleProps}>
      <Card.RichMedia>
        {ExampleCardImage01}
      </Card.RichMedia>
      <Flex alignItems="center" p={exampleCardPadding}>
        {ExampleCardThumbnail01}
        <Box pl={exampleCardPadding}>
          <Card.Title>
            Card title
          </Card.Title>
          <Card.Subtitle>
            Secondary text
          </Card.Subtitle>
        </Box>
      </Flex>
      <Card.Body pb={exampleCardPadding} px={exampleCardPadding}>
        {ExampleCardBody01}
      </Card.Body>
      <Card.Actions justifyContent="space-between" px={exampleCardPadding}>
        {ExampleCardActions01}
        {ExampleCardActions02}
      </Card.Actions>
    </Card>
  </Box>
  <Box {...baseBoxStyleProps}>
    <CodeBlock code="// Passed in through Props" />
    <Card
      cardActions={ExampleCardActions01}
      cardActionsRight={ExampleCardActions02}
      cardBody={ExampleCardBody01}
      cardSecondaryMedia={ExampleCardImage01}
      cardSubtitle="Secondary text"
      cardThumbnail={ExampleCardThumbnail01}
      cardTitle="Card title"
      {...baseCardStyleProps}
    />
  </Box>
  <Box {...baseBoxStyleProps}>
    <CodeBlock code="// Using subComponents" />
    <Card cardPadding={0} pb={exampleCardPadding} {...baseCardStyleProps}>
      <Flex alignItems="center" p={exampleCardPadding}>
        {ExampleCardThumbnail01}
        <Box pl={exampleCardPadding}>
          <Card.Title>
            Card title
          </Card.Title>
          <Card.Subtitle>
            Secondary text
          </Card.Subtitle>
        </Box>
      </Flex>
      <Card.SecondaryMedia>
        {ExampleCardImage02}
      </Card.SecondaryMedia>
      <Card.Body p={exampleCardPadding}>
        {ExampleCardBody01}
      </Card.Body>
      <Card.Actions justifyContent="space-between" px={exampleCardPadding}>
        {ExampleCardActions01}
        {ExampleCardActions02}
      </Card.Actions>
    </Card>
  </Box>
</Flex>
```

#### Hide Card

```jsx inside Markdown
import { useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { Card } from 'src/elements/Card';
import { Text } from 'src/elements/typography';
import {
  Box,
  Flex,
} from 'src/elements/grid';
import { Button } from 'src/elements/Button';

const [hide, setHidden] = useState(false);

<ThemeProvider theme={{}}>
  <Button
    onClick={() => setHidden(!hide)}
  >
    Toggle Visibility
  </Button>
  <Card isHidden={hide}>
    <Card.Body>
      <Flex flexDirection={['column', 'row']}>
        <Box px={1}>
        <Text>
          Hideable
        </Text>
        </Box>
      </Flex>
    </Card.Body>
  </Card>
</ThemeProvider>
```
