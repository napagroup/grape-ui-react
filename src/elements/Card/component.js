import React from 'react';
import { Box } from 'src/elements/grid';
import { removeSomeProps } from 'src/utils/componentHelpers';
import {
  cardDefaultProps,
  cardPropTypes,
} from './utils';
import {
  CardHeader,
  CardInner,
  getCardActions,
  getCardBody,
  getCardRichMedia,
  getCardSecondaryMedia,
} from './subComponents';

export const CardComponent = props => {
  const {
    cardPadding,
    cardSecondaryMedia,
    children,
  } = props;
  return (
    <Box {...removeSomeProps(props, Object.keys(cardPropTypes))}>
      {getCardRichMedia(props)}
      <CardInner {...props}>
        <CardHeader {...props} />
        {!cardSecondaryMedia ? getCardBody(props) : ''}
        {children ? <Box p={cardPadding}>{children}</Box> : ''}
      </CardInner>
      {getCardSecondaryMedia(props)}
      {getCardActions(props)}
    </Box>
  );
};

CardComponent.propTypes = {
  ...cardPropTypes,
};

CardComponent.defaultProps = {
  ...cardDefaultProps,
};
