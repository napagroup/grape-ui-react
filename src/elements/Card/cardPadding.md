The `<Card>` component is meant to be easy for quick designs that match Material's guidelines, while still allowing for styled-system's built in styling functions.  When defining the base spacing settings for your `<Card>`, please use the prop `cardPadding` to define props.  This will assure that al of the additional prop provided areas having the proper spacing.

**Note:** if you are using `<Card>` via subComponents, please default `cardPadding: 0`.

#### `cardPadding` Example

```jsx is Markdown
import { Button } from 'src/elements/Button';
import { Card } from 'src/elements/Card';
import { Box, Flex } from 'src/elements/grid';
import { Image } from 'src/elements/Image';
import { CodeBlock, Paragraph } from 'src/elements/typography';

const getRandomBill = (min, max, ratio) => {
  const numberX = Math.floor(Math.random() * (max - min) + min);
  const numberY = Math.floor(numberX / ratio);
  return `http://www.fillmurray.com/${numberX}/${numberY}`;
}

const ExampleActions01 = (
  <div>
    <Button bg="brandPrimary" bgHoverColor="brandPrimary.light" color="white" m={0} mr={1}>Action</Button>
    <Button bg="brandPrimary" bgHoverColor="brandPrimary.light" color="white" m={0} ml={1}>Action</Button>
  </div>
);

const ExampleActions02 = (
  <Button m={0}>⛳️</Button>
);

const ExampleBody01 = (
  <Paragraph mb={0}>
    Bacon ipsum dolor amet drumstick pork chop cupim tenderloin picanha short ribs landjaeger chicken, strip steak venison.
  </Paragraph>
);

const ExampleCardImage01 = (
  <Image
    alt="Placeholder photo from FillMurray.com"
    display="block" // be sure to mark this as block or else the image can cause unwanted spacing on the bottom of the card
    src={getRandomBill(400, 1000, 4 / 3)}
  />
);

const ExampleThumbnail01 = (
  <Image
    alt="Placeholder photo from FillMurray.com"
    borderRadius="50%"
    display="block" // be sure to mark this as block or else the image can cause unwanted spacing on the bottom of the card
    src={getRandomBill(40, 100, 1 / 1)}
    width={40}
  />
);

const baseCardStyleProps = {
  m: 1,
  maxWidth: 300,
  width: 1,
};

const containerStyleProps = {
  alignItems: 'center',
  background: '#fdf4fb',
  flexDirection: ['column', 'row'],
  justifyContent: 'space-evenly',
  p: [2, null, 3],
};

const codeProps = `// Passed in through Props
// cardPadding using default value
`;

const codeSub = `// Using subComponents
// cardPadding: 0
`;

<Flex {...containerStyleProps}>
  <Box {...baseCardStyleProps}>
    <CodeBlock code={codeProps} />
    <Card
      cardActions={ExampleActions01}
      cardActionsRight={ExampleActions02}
      cardBody={ExampleBody01}
      cardSecondaryMedia={ExampleCardImage01}
      cardSubtitle="Secondary text"
      cardThumbnail={ExampleThumbnail01}
      cardTitle="Card title"
      {...baseCardStyleProps}
    />
  </Box>
  <Box {...baseCardStyleProps}>
    <CodeBlock code={codeSub} />
    <Card cardPadding={0}  pb={3} {...baseCardStyleProps}>
      <Flex alignItems="center" p={3}>
        <Card.Thumbnail>
          {ExampleThumbnail01}
        </Card.Thumbnail>
        <Box pl={3}>
          <Card.Title>
            Card title
          </Card.Title>
          <Card.Subtitle>
            Secondary text
          </Card.Subtitle>
        </Box>
      </Flex>
      <Card.SecondaryMedia>
        {ExampleCardImage01}
      </Card.SecondaryMedia>
      <Card.Body p={3}>
        {ExampleBody01}
      </Card.Body>
      <Card.Actions justifyContent="space-between" px={3}>
        {ExampleActions01}
        {ExampleActions02}
      </Card.Actions>
    </Card>
  </Box>
</Flex>
```
