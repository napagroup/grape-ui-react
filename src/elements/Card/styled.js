import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import propTypes from '@styled-system/prop-types';
import { withHideable } from 'src/elements/utils/Hideable';
import { Box, Flex } from 'src/elements/grid';
import { Header, Paragraph } from 'src/elements/typography';
import { CardComponent } from './component';
import { cardPropTypes, cardDefaultProps } from './utils';

const Card = styled(withHideable(CardComponent))``;

Card.propTypes = {
  ...cardPropTypes,
};

Card.defaultProps = {
  ...cardDefaultProps,
};

const subComponentStarter = [
  {
    component: Flex,
    componentDefaultProps: {},
    componentPropTypes: {},
    name: 'Actions',
  },
  {
    component: Box,
    componentDefaultProps: {},
    componentPropTypes: {},
    name: 'Body',
  },
  {
    component: Box,
    componentDefaultProps: {},
    componentPropTypes: {},
    name: 'Inner',
  },
  {
    component: Box,
    componentDefaultProps: {
      width: 1,
    },
    componentPropTypes: {
      ...propTypes.layout,
    },
    name: 'RichMedia',
  },
  {
    component: Box,
    componentDefaultProps: {
      width: 1,
    },
    componentPropTypes: {
      ...propTypes.layout,
    },
    name: 'SecondaryMedia',
  },
  {
    component: Paragraph,
    componentDefaultProps: {
      color: 'gray',
      mb: 0,
      sm: true,
    },
    componentPropTypes: {
      ...propTypes.space,
      color: PropTypes.string,
      sm: PropTypes.bool,
    },
    name: 'Subtitle',
  },
  {
    component: Box,
    componentDefaultProps: {},
    componentPropTypes: {},
    name: 'Thumbnail',
  },
  {
    component: Header.h5,
    componentDefaultProps: {
      mb: 0,
    },
    componentPropTypes: {
      ...propTypes.space,
    },
    name: 'Title',
  },
];

subComponentStarter.forEach(starter => {
  const {
    name,
    componentDefaultProps,
    componentPropTypes,
    component: SubComponent,
  } = starter;
  Card[name] = props => {
    // eslint-disable-next-line react/prop-types
    const { children } = props;
    return (
      <SubComponent {...props}>{children}</SubComponent>
    );
  };
  Card[name].propTypes = {
    ...componentPropTypes,
    children: PropTypes.node,
  };
  Card[name].defaultProps = {
    ...componentDefaultProps,
    children: '',
  };
});

/** @component */
export { Card };
