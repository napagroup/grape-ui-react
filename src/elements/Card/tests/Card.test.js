import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Card } from 'src/elements/Card';

const emptyTheme = {};

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};

describe('Box Component base', () => {
  it('should return a Card with base styling', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Card
          cardActions="cardActions"
          cardActionsLeft="cardActionsLeft"
          cardActionsRight="cardActionsRight"
          cardBody="cardBody"
          cardRichMedia="cardRichMedia"
          cardSecondaryMedia="cardSecondaryMedia"
          cardSubtitle="cardSubtitle"
          cardThumbnail="cardThumbnail"
          cardTitle="cardTitle"
        >
          I am a child.
        </Card>
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Card with nuances', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Card
          cardActionsRight="Card Actions Right"
          cardBody="Card Body"
        />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Card with objects in title', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Card
          cardSubtitle={<div />}
          cardTitle={<div />}
        />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return an empty Card', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Card />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
