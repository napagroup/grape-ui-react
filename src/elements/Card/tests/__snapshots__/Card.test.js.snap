// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component base should return a Card with base styling 1`] = `
.c6 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.25rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

.c7 {
  color: hsl(0,0%,50%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

.c0 {
  box-sizing: border-box;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
  margin-bottom: 4px;
}

.c1 {
  box-sizing: border-box;
  width: 100%;
}

.c2 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
}

.c4 {
  box-sizing: border-box;
  padding-right: 16px;
}

.c5 {
  box-sizing: border-box;
}

.c8 {
  box-sizing: border-box;
  padding: 16px;
}

.c9 {
  box-sizing: border-box;
  padding-bottom: 16px;
}

.c10 {
  box-sizing: border-box;
  width: 100%;
  padding-bottom: 16px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: 16px;
  padding-top: 16px;
}

.c11 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    margin-bottom: 8px;
  }
}

<div
  className="c0 "
>
  <div
    className="c1"
  >
    cardRichMedia
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        cardThumbnail
      </div>
      <div
        className="c5"
      >
        <h5
          className="c6"
        >
          cardTitle
        </h5>
        <p
          className="c7"
        >
          cardSubtitle
        </p>
      </div>
    </div>
    
    <div
      className="c8"
    >
      I am a child.
    </div>
  </div>
  <div
    className="c9"
  >
    <div
      className="c10"
    >
      cardSecondaryMedia
    </div>
    <div
      className="c2"
    >
      cardBody
    </div>
  </div>
  <div
    className="c11"
  >
    cardActions
    <div
      className="c5"
    >
      cardActionsLeft
    </div>
    <div
      className="c5"
    >
      cardActionsRight
    </div>
  </div>
</div>
`;

exports[`Box Component base should return a Card with nuances 1`] = `
.c0 {
  box-sizing: border-box;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
  margin-bottom: 4px;
}

.c1 {
  box-sizing: border-box;
  padding-bottom: 16px;
  padding-top: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

.c3 {
  box-sizing: border-box;
}

.c4 {
  box-sizing: border-box;
  padding-top: 16px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: 16px;
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    margin-bottom: 8px;
  }
}

<div
  className="c0 "
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      
      <div
        className="c3"
      >
        
        
      </div>
    </div>
    <div
      className="c4"
    >
      Card Body
    </div>
    
  </div>
  <div
    className="c5"
  >
    <div
      className="c3"
    >
      Card Actions Right
    </div>
  </div>
</div>
`;

exports[`Box Component base should return a Card with objects in title 1`] = `
.c0 {
  box-sizing: border-box;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
  margin-bottom: 4px;
}

.c1 {
  box-sizing: border-box;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    margin-bottom: 8px;
  }
}

<div
  className="c0 "
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      
      <div
        className="c3"
      >
        <div
          className="c3"
        >
          <div />
        </div>
        <div
          className="c3"
        >
          <div />
        </div>
      </div>
    </div>
    
  </div>
  
</div>
`;

exports[`Box Component base should return an empty Card 1`] = `
.c0 {
  box-sizing: border-box;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
  margin-bottom: 4px;
}

.c1 {
  box-sizing: border-box;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    margin-bottom: 8px;
  }
}

<div
  className="c0 "
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      
      <div
        className="c1"
      >
        
        
      </div>
    </div>
    
  </div>
  
</div>
`;
